homeassistant/components/recorder/history.py

```diff
@@ -478,11 +479,10 @@ def get_states(
     no_attributes: bool = False,
 ) -> list[State]:
     """Return the states at a specific point in time."""
-    if (
-        run is None
-        and (run := (recorder.run_information_from_instance(hass, utc_point_in_time)))
-        is None
-    ):
+    if run is None:
+        run = recorder.get_instance(hass).run_history.get(utc_point_in_time)
```

Comment:
This call was previously incorrect because it would only look at the current run

Reply:
Realized that this is never called in production code anymore (nor is `get_state`)

I'm planning on removing these apis in a followup

Reply:
https://github.com/home-assistant/core/pull/70830